### PR TITLE
[WIP] Add Mirage Raceway Race Event

### PR DESCRIPTION
**Pull request by NickTyrer**
Continuation of @alexkulya's work
https://github.com/vmangos/core/issues/658

### CODE REVIEW

File: sql/migrations/20210408151914_world.sql [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+(4251*100+02, 4251, 0, 11, 0, 100, 0, 0, 0, 0, 0, 4251*100+04, 0, 0, "Goblin Racer - Just Spawned - Start Script");
+
+DELETE FROM `creature_ai_scripts` WHERE `id` IN (4251*100+01, 4251*100+02, 4251*100+03, 4251*100+04);
+INSERT INTO `creature_ai_scripts` (`id`, `delay`, `command`, `datalong`, `datalong2`, `datalong3`, `datalong4`, `target_param1`, `target_param2`, `target_type`, `data_flags`, `dataint`, `dataint2`, `dataint3`, `dataint4`, `x`, `y`, `z`, `o`, `condition_id`, `comments`) VALUES
+(4251*100+01, 0, 15, 6600, 0, 0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, "Goblin Racer - Cast Spell 'Salt Flats Racer Speed'"),
+(4251*100+02, 0, 15, 6601, 0, 0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, "Goblin Racer - Cast Spell 'Salt Flats Racer Slow'"),
+(4251*100+03, 0, 15, 6602, 0, 0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, "Goblin Racer - Cast Spell 'Salt Flats Racer Normal'"),
+(4251*100+04, 0, 39, 4251*100+04, 0, 0, 0, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 0, 0, 0, "Goblin Racer - Start Event");
+
+DELETE FROM `event_scripts` WHERE `id` = 4251*100+04;
```
Comment (ratkosrb) [inline sql/migrations/20210408151914_world.sql]: I believe you actually want to use generic_scripts, not event_scripts here.

**Comment by sdpHTF9173** [comment]
Excuse me，I've updated this file, but the event can't start yet，Does this script files need a cpp file to start this event?  i see it working in progress。

**Comment by alexkulya** [comment]
@sdpHTF9173 At the moment when I wrote the script for this event, it worked, but with some shortcomings, later some functions in the core changed, event scripts and at the moment the event is completely not working.

**Comment by sdpHTF9173** [comment]
> At the moment when I wrote the script for this event, it worked, but with some shortcomings, later some functions in the core changed, event scripts and at the moment the event is completely not working.

Thanks，Looking forward to the early completion！：）

**Comment by sdpHTF9173** [comment]
Looking forward to  Mirage Raceway Race Event completion！：）

**Comment by sdpHTF9173** [comment]
Looking forward to Mirage Raceway Race Event completion！+2

**Comment by sdpHTF9173** [comment]
WoWScrnShot_080421_174836
WoWScrnShot_080421_174905
WoWScrnShot_080421_174934
WoWScrnShot_080421_174956
WoWScrnShot_080421_175008
WoWScrnShot_080421_175303
WoWScrnShot_080421_175312
This script seems to work whenever I restart the server，But it can only happen once, and the goblins win，hope it can be improved as soon as possible。@alexkulya @NickTyrer

**Comment by sdpHTF9173** [comment]
Looking forward to Mirage Raceway Race Event completion！+3

**Comment by sdpHTF9173** [comment]
Looking forward to Mirage Raceway Race Event completion！+4

**Pull request closed without merging**